src/modules/hyprland/workspaces.cpp

```diff
@@ -43,68 +53,92 @@ Workspaces::Workspaces(const std::string &id, const Bar &bar, const Json::Value
   gIPC->registerForIPC("workspace", this);
   gIPC->registerForIPC("createworkspace", this);
   gIPC->registerForIPC("destroyworkspace", this);
+  gIPC->registerForIPC("focusedmon", this);
+  gIPC->registerForIPC("moveworkspace", this);
 }
 
 auto Workspaces::update() -> void {
-  for (int &workspace_to_remove : workspaces_to_remove_) {
+  for (std::string workspace_to_remove : workspaces_to_remove_) {
     remove_workspace(workspace_to_remove);
   }
 
   workspaces_to_remove_.clear();
 
-  for (int &workspace_to_create : workspaces_to_create_) {
+  for (Json::Value &workspace_to_create : workspaces_to_create_) {
     create_workspace(workspace_to_create);
   }
 
   workspaces_to_create_.clear();
 
-  for (std::unique_ptr<Workspace> &workspace : workspaces_) {
-    workspace->set_active(workspace->id() == active_workspace_id);
-
+  for (auto &workspace : workspaces_) {
+    workspace->set_active(workspace->name() == active_workspace_name);
     std::string &workspace_icon = icons_map_[""];
     if (with_icon_) {
       workspace_icon = workspace->select_icon(icons_map_);
     }
-
     workspace->update(format_, workspace_icon);
   }
-
   AModule::update();
 }
 
 void Workspaces::onEvent(const std::string &ev) {
   std::lock_guard<std::mutex> lock(mutex_);
   std::string eventName(begin(ev), begin(ev) + ev.find_first_of('>'));
   std::string payload = ev.substr(eventName.size() + 2);
+
   if (eventName == "workspace") {
-    std::from_chars(payload.data(), payload.data() + payload.size(), active_workspace_id);
+    active_workspace_name = payload;
+
   } else if (eventName == "destroyworkspace") {
-    int deleted_workspace_id;
-    std::from_chars(payload.data(), payload.data() + payload.size(), deleted_workspace_id);
-    workspaces_to_remove_.push_back(deleted_workspace_id);
+    workspaces_to_remove_.push_back(payload);
+
   } else if (eventName == "createworkspace") {
-    int new_workspace_id;
-    std::from_chars(payload.data(), payload.data() + payload.size(), new_workspace_id);
-    workspaces_to_create_.push_back(new_workspace_id);
+    const Json::Value workspaces_json = gIPC->getSocket1JsonReply("workspaces");
+    for (Json::Value workspace_json : workspaces_json) {
+      if (workspace_json["name"].asString() == payload &&
+          (all_outputs() || bar_.output->name == workspace_json["monitor"].asString()) &&
+          (workspace_json["name"].asString().find("special:") != 0 || show_special())) {
```

Comment:
Let's put `show_special` check at first in order to skip comparing strings if flag is set to true.

Reply:
And on 155 line too.

Reply:
done